ArduCopter/mode_auto.cpp

```diff
@@ -714,6 +740,11 @@ bool ModeAuto::verify_command(const AP_Mission::Mission_Command& cmd)
 //      called by auto_run at 100hz or more
 void ModeAuto::takeoff_run()
 {
+    // if the user doesn't want to raise the throttle we can set it automatically
+    // note that this can defeat the disarm check on takeoff
+    if ((copter.g2.auto_options & (int32_t)Options::BypassThrottle) != 0) {
```

Comment:
The problem with that is that if you combine the bypass throttle check with the allow arming check. The mission will be running before arming, and the normal call to `update_auto_armed()` will set this to false because the motors aren't armed. If we run it once on takeoff you can't combine the allow arming and bypass, whereas by continually flagging it during the takeoff phase of flight we are able to ensure it actually takesoff.